SU2_CFD/include/solver_structure.hpp

```diff
@@ -509,7 +528,51 @@ class CSolver {
    * \param[in] config - Definition of the particular problem.
    */
   void SetAuxVar_Surface_Gradient(CGeometry *geometry, CConfig *config);
+
+  /*!
+   * \brief Set the solution vector to solution in Solution_Old.
+   * \param[in] geometry - The geometrical definition of the problem.
+   * \param[in] config - The particular config.
+   */
+  void SetSolution_Old(CGeometry *geometry);
   
+  /*!
+   * \brief Add External_Old to Solution vector.
+   * \param[in] geometry - The geometrical definition of the problem.
+   * \param[in] config - The particular config.
+   */
+  void AddSolution_ExternalOld(CGeometry *geometry);
+
+  /*!
+   * \brief Set the Solution vector to zero.
+   * \param[in] geometry - The geometrical definition of the problem.
+   */
+  void SetSolution_Zero(CGeometry *geometry);
+
+  /*!
+   * \brief Set the External vector to zero.
+   * \param[in] geometry - The geometrical definition of the problem.
+   */
+  void SetExternal_Zero(CGeometry *geometry);
+
+  /*!
+   * \brief Add the current Solution vector to External.
+   * \param[in] geometry - The geometrical definition of the problem.
+   */
+  void AddExternal_Solution(CGeometry *geometry);
+
+  /*!
+   * \brief Add the current Solution vector to External_Old.
+   * \param[in] geometry - The geometrical definition of the problem.
+   */
+  void AddExternalOld_Solution(CGeometry *geometry);
```

Comment:
That's the one (`Set_OldExternal`) that does not fit this structure. I wanted to have the same naming as for `Solution`.
Concerning the underscores in all the other routines: I wanted to give them at least some sense (the variable preceding the underscore being the one that is affected, the variable following the underscore being the input value). They are often used quite randomly or there's a different rule that I haven't discovered yet :)

Reply:
I've not noticed a pattern yet either. That is why I suggested stating everything (operation Add/Set, source, and destination) in the name. I get your convention but since like you say other methods do not seem to follow one, the best option might be to spell everything.
It may avoid future situation where someone comes along and creates a redundant method (that seems to have happened in CVariable).

Reply:
I made some names more clear and kept some for which there is a de facto convention.